packages/svelte/src/compiler/phases/1-parse/state/tag.js

```diff
@@ -577,7 +577,12 @@ function special(parser) {
 
 		const expression = read_expression(parser);
 
-		if (expression.type !== 'CallExpression' || expression.callee.type !== 'Identifier') {
+		if (
+			expression.type !== 'CallExpression' &&
+			(expression.type !== 'ChainExpression' ||
+				expression.expression.type !== 'CallExpression' ||
+				!expression.expression.optional)
```

Comment:
Does this mean that I can do `snippet?.()` but not `snippets.snippet?.()`? If so this might need to be recursive?

Reply:
It doesn't, the AST is a bit weird in that regard, it's always a callexpression or a chainexpression with a callexpression for anything that ends with a function call.